nixos/modules/services/web-apps/c2FmZQ-server.nix

```diff
@@ -0,0 +1,137 @@
+{ lib, pkgs, config, ... }:
+
+let
+  inherit (lib) mkEnableOption mkPackageOption mkOption types;
+
+  cfg = config.services.c2FmZQ-server;
+
+  argsFormat = with lib; {
+    type = with types; let
+      valueType = nullOr (oneOf [
+        bool int str
+      ]) // {
+        description = "c2FmZQ-server params";
+      };
+    in valueType;
+    generate = name: value:
+      let mkParam = k: v:
+            if v == null then []
+            else if isBool v then optional v ("--"+k)
+            else [("--"+k) v];
+      in escapeShellArgs (builtins.concatLists (mapAttrsToList mkParam value));
+  };
+in {
+  options.services.c2FmZQ-server = {
+    enable = mkEnableOption "c2FmZQ-server";
+
+    bindIP = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description =  "The local address to use.";
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8080;
+      description =  "The local port to use.";
+    };
+
+    passphraseFile = mkOption {
+      type = types.str;
```

Comment:
```suggestion
      type = types.str;
      example = ...
```

Reply:
Not sure if the `toFile` example in the test or `"/run/secrets/c2FmZQ/pwfile"` should be used